test/OpenTelemetry.Resources.OperatingSystem.Tests/OperatingSystemDetectorTests.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Xunit;
+
+namespace OpenTelemetry.Resources.OperatingSystem.Test;
+
+public class OperatingSystemDetectorTests
+{
+    [Fact]
+    public void TestOperatingSystemAttributes()
+    {
+        var resource = ResourceBuilder.CreateEmpty().AddDetector(new OperatingSystemDetector()).Build();
```

Comment:
I'd suggest to use the extension method instead of using the ctor, so as to mimic the user code more closely, and also this can help remove the InternalsVisible trick as well. There is no need for Tests to know any internals.

Reply:
Won't be a bad idea at all to copy the semantic convention OR add ref to it from test project.

Reply:
@cijothomas, in general I agree, but current approach is in sync with all other projects.
If we want to change lets do it in separate PR/issue.

Reply:
yes agree, no need to block this PR at all.